public/app/features/alerting/unified/components/rule-viewer/RuleViewer.tsx

```diff
@@ -274,33 +285,47 @@ export const Title = ({ name, paused = false, state, health, ruleType, ruleOrigi
  */
 function PrometheusConsistencyCheck({ ruleIdentifier }: { ruleIdentifier: RuleIdentifier }) {
   const [ref, { width }] = useMeasure<HTMLDivElement>();
-  const { isConsistent, error } = usePrometheusCreationConsistencyCheck(ruleIdentifier);
 
-  if (isConsistent) {
-    return null;
-  }
+  const { result: ruleLocation } = useRuleLocation(ruleIdentifier);
+  const { waitForGroupConsistency, groupConsistent } = useRuleGroupConsistencyCheck();
+
+  const [waitAction, waitState] = useAsync((groupIdentifier: RuleGroupIdentifierV2) => {
+    return waitForGroupConsistency(groupIdentifier);
+  });
 
-  if (error) {
+  useEffect(() => {
+    if (ruleLocation) {
+      waitAction.execute(ruleLocation.groupIdentifier);
+    }
+  }, [ruleLocation, waitAction]);
+
+  if (isError(waitState)) {
     return (
       <Alert title="Unable to check the rule status" bottomSpacing={0} topSpacing={2}>
-        {stringifyErrorLike(error)}
+        {stringifyErrorLike(waitState.error)}
       </Alert>
     );
   }
 
-  return (
-    <Stack direction="column" gap={0} ref={ref}>
-      <LoadingBar width={width} />
-      <Alert
-        title={t('alerting.rule-viewer.prometheus-consistency-check.alert-title', 'Update in progress')}
-        severity="info"
-      >
-        <Trans i18nKey="alerting.rule-viewer.prometheus-consistency-check.alert-message">
-          Alert rule has been updated. Changes may take up to a minute to appear on the Alert rules list view.
-        </Trans>
-      </Alert>
-    </Stack>
-  );
+  // If groupConsistent is undefined, it means that the rule is still being checked and we don't know if it's consistent or not
```

Comment:
this comment should be If `groupConsistent` is `false` ? or the condition should be` if (groupConsistent === undefined) {` ?

Reply:
The comment was a bit misleading, I've updated it 😄